components/usage/pkg/stripe/stripe.go

```diff
@@ -320,6 +330,83 @@ func (c *Client) GetSubscriptionWithCustomer(ctx context.Context, subscriptionID
 	return subscription, nil
 }
 
+func (c *Client) CreateSubscription(ctx context.Context, customerID string, priceID string, isAutomaticTaxSupported bool) (*stripe.Subscription, error) {
+	if customerID == "" {
+		return nil, fmt.Errorf("no customerID specified")
+	}
+	if priceID == "" {
+		return nil, fmt.Errorf("no priceID specified")
+	}
+
+	startOfNextMonth := getStartOfNextMonth()
+
+	params := &stripe.SubscriptionParams{
+		Customer: &customerID,
+		Items: []*stripe.SubscriptionItemsParams{
+			{
+				Price: &priceID,
+			},
+		},
+		AutomaticTax: &stripe.SubscriptionAutomaticTaxParams{
+			Enabled: &isAutomaticTaxSupported,
+		},
+		BillingCycleAnchor: stripe.Int64(startOfNextMonth.Unix()),
+	}
+
+	subscription, err := c.sc.Subscriptions.New(params)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get subscription with customer ID %s", customerID)
+	}
+
+	return subscription, err
+}
+
+func getStartOfNextMonth() time.Time {
+	now := time.Now()
+	currentYear, currentMonth, _ := now.Date()
+
+	firstOfMonth := time.Date(currentYear, currentMonth, 1, 0, 0, 0, 0, time.UTC)
+	startOfNextMonth := firstOfMonth.AddDate(0, 1, 0)
+
+	return startOfNextMonth
+}
```

Comment:
A more general version (and also a more testable) is
```suggestion
func getStartOfNextMonth(t time.Time) time.Time {
	currentYear, currentMonth, _ := t.Date()

	firstOfMonth := time.Date(currentYear, currentMonth, 1, 0, 0, 0, 0, time.UTC)
	startOfNextMonth := firstOfMonth.AddDate(0, 1, 0)

	return startOfNextMonth
}
```

Reply:
How is it then used in the test? I did `getStartOfNextMonth(time.Time{})` but I still got `not enough arguments in call to getStartOfNextMonth`

Reply:
You pass in a time, and get back start of next month. So for example:

```golang
ts := time.Date(2022, 10, 17, 10, 10, 10, time.UTC)
getStartOfNextMonth(ts)
require.Equal(t, time.Date(2022, 11, 17, 10, 10, 10, time.UTC)
```

Reply:
You're likely shadowing the `t testing.T` variable